codetransformer/code.py

```diff
@@ -0,0 +1,432 @@
+            elif instr.absjmp:
+                # Resolve the absolute jump target.
+                bc.extend(self.sparse_index(instr.arg).to_bytes(2, 'little'))
+            elif instr.reljmp:
+                # Resolve the relative jump target.
+                # We do this by subtracting the curren't instructions's
+                # sparse index from the sparse index of the argument.
+                # We then subtract 3 to account for the 3 bytes the
+                # current instruction takes up.
+                bc.extend((
+                    self.sparse_index(instr.arg) - self.sparse_index(instr) - 3
+                ).to_bytes(2, 'little',))
+            elif instr.have_arg:
+                # Write any other arg here.
+                bc.extend(instr.arg.to_bytes(2, 'little'))
+
+        return CodeType(
+            self.argcount,
+            self.kwonlyargcount,
+            len(varnames),
+            max(
+                scanl(
+                    operator.add,
+                    0,
+                    map(operator.attrgetter('stack_effect'), self.instrs)
+                ),
+            ),
+            self.flags,
+            bytes(bc),
+            consts,
+            names,
+            varnames,
+            self.filename,
+            self.name,
+            self.firstlineno,
+            self.lnotab,
+            freevars,
+            cellvars,
+        )
+
+    @property
+    def instrs(self):
+        return self._instrs
+
+    @property
+    def sparse_instrs(self):
+        return tuple(_sparse_args(self.instrs))
+
+    @property
+    def argcount(self):
+        return self._argcount
+
+    @property
+    def kwonlyargcount(self):
+        return self._kwonlyargcount
+
+    @property
+    def consts(self):
+        # We cannot use a set comprehension because consts do not need
+        # to be hashable.
+        consts = []
+        append_const = consts.append
+        for instr in self.instrs:
+            if isinstance(instr, LOAD_CONST) and instr.arg not in consts:
+                append_const(instr.arg)
+        return tuple(consts)
+
+    @property
+    def names(self):
+        return tuple(sorted({
+            instr.arg for instr in self.instrs if instr.uses_name
+        }))
+
+    @property
+    def argnames(self):
+        return self._argnames
+
+    @property
+    def varnames(self):
+        return self._argnames + tuple(sorted({
+            instr.arg for instr in self.instrs if instr.uses_varname
+        }))
+
+    @property
+    def cellvars(self):
+        return self._cellvars
+
+    @property
+    def freevars(self):
+        return self._freevars
+
+    @property
+    def flags(self):
+        return self._flags
+
+    @property
+    def nested(self):
+        return bool(self._flags & Flags.CO_NESTED)
+
+    @property
+    def generator(self):
+        return bool(self._flags & Flags.CO_GENERATOR)
+
+    @property
+    def coroutine(self):
```

Comment:
I wonder whether there's an argument for having `coroutine` be true for iterable coroutines as well?